### ISSUE TITLE
[Feature Request] Flag a request to be exclude from test runner

### ISSUE THREAD

**Issue by stgonzales** [usebruno/bruno#1039]
It would be good to have a way to mark each request to be exclude from the test runner. This could be by adding a meta flag or in a blacklist format.

**Comment by helloanoop** [comment]
Yes, we will support this.

I am thinking of supporting a more generic feature called `tags`
Here is how I am envisioning it

- A tag is essential a string. Ex: `regression`, `sanity`
- Each request can have multiple tags
- When running the collection, we should be able to run tests where only requests with a certain set of tags get included

Here is how it would look in the bru file.

```groovy
meta {
  name: Get Users
  type: http
  seq: 1,
  tags: [
    sanity,
    regression
  ]
}
```

The GUI Collection runner will support options to only run requests which match a certain set of tags.

In the CLI, we can have something like
```bash
bru run --tags=sanity,regression
```

Let me know if you have feedback or suggestions

cc @dcoomber @jeff-edmondson-pci @rabestro @mj-h @Its-treason @martinsefcik

**Comment by rabestro** [comment]
@helloanoop 
The concept of tags is handy, and it is supported by many testing frameworks. I suggest considering the option not only to run tests that contain a specific tag but also to exclude tests with a particular tag. For example, in the Karate framework, there is no special annotation, but you can add a tag named "ignore" and then run tests while excluding the ones with the "ignore" tag.

**Comment by KirkMunroSagent** [comment]
> @helloanoop The concept of tags is handy, and it is supported by many testing frameworks. I suggest considering the option not only to run tests that contain a specific tag but also to exclude tests with a particular tag. For example, in the Karate framework, there is no special annotation, but you can add a tag named "ignore" and then run tests while excluding the ones with the "ignore" tag.

Maybe something like this then?

`bru run --tags=sanity,regression --exclude-tags=ignore`

That way it's not specific to a hard-coded tag name like `ignore`, and the collection that is run is the aggregate of all tests that match at least one of the tags in the `--tags` argument, minus any tests that match at least one of the tags in the `--exclude-tags` list.

**Comment by brvaland** [comment]
@helloanoop do we have any estimate when this feature will be available as we integrate our collection with CICD and would like to exclude few POST request for PRODUCTION environment to avoid adding data.

Is there a workaround which I can explore via script to move to next/skip request in the runner.?

**Comment by irharrier2** [comment]
@helloanoop your suggested solution is perfect. Any idea when are you going to deliver it?

**Comment by nikischin** [comment]
Seems to be related to or possibly even implemented (different solution than suggested) with #2397

**Comment by Aquil0214** [comment]
I am eagerly waiting for this feature but I want to unblock myself, if possible kindly provide any alternative solution.

**Comment by Ram3812** [comment]
I'm looking forward to the availability of this feature. Could you please share any expected timeline for it on both Runner and CLI.?

**Comment by nikischin** [comment]
Should be pretty much implemented with #2716 I guess?

**Comment by anusree-bruno** [comment]
@Ram3812  as @nikischin  mentioned we have added feature to add tags for requests in v2.8.0

In the cli you can use this command to include or exclude requests

`--tags=tag_1,tag_2 --exclude-tags=tag_3`

**Comment by Ram3812** [comment]
@nikischin & @anusree-bruno 
I'm able to use the "--tags=tag_1,tag_2 --exclude-tags=tag_3" commands by updating to v2.8.0.
It would be more helpful if the documentation is also updated. (https://docs.usebruno.com/bru-cli/commandOptions).

Thanks.

**Comment by anusree-bruno** [comment]
yes @Ram3812 we are working on the documnets on all v2.8.0 feature updates.

**Comment by anusree-bruno** [comment]
We’ve updated our documentation to cover the following topics:

[Controlling Request Execution with Tags (Manual Test)](https://docs.usebruno.com/testing/automate-test/manual-test#controlling-request-execution-with-tags)

[Request Options in BRU CLI](https://docs.usebruno.com/bru-cli/commandOptions#request-options)

Additionally, we're introducing the ability to select or deselect and reorder requests before running them. This will be available in next release
